automatic/freecad/update_helper.ps1

```diff
@@ -3,16 +3,33 @@ function Get-FreeCad {
 param(
   [string]$Title,
   [string]$kind = 'stable',
-  [int]$filler = "0"
+  [int]$filler = "0",
+  [string]$uri = $releases
 )
-  $mobile = @{$true="portable";$false="installer"}[($kind -eq 'portable')]; $portable = @{$true="vc14.x-x86_64";$false="$mobile"}[($kind -eq 'dev')]
+  $download_page = Invoke-WebRequest -Uri $uri -UseBasicParsing
+  $mobile = @{$true="portable";$false="installer"}[($kind -eq 'portable')]; $portable = @{$true="vc(17|14)\.x\-x86(\-|_)64";$false="$mobile"}[($kind -eq 'dev')]
   $try = (($download_page.Links | ? href -match "CAD\-|\.[\dA-Z]+\-WIN" | select -First 1 -expand href).Split("/")[-1]).Split(".")[-1]
   $ext = @{$true='7z';$false='exe'}[( $kind -match 'dev' ) -or ( $portable -match 'portable' )]
   $re32 = "(WIN)\-x32\-($portable)\.$ext";
-  $re64 =  @{$true="(WIN|Win)\-(Conda|x64)(\-|_)$portable+(\.$ext)$";$false="(WIN)\-x64\-($portable)(\-[0-9])?\.$ext$"}[( $kind -match 'dev' )]
+  # cybrwshl use of (\-[0-9])? regex updated to (\-\d+)? for installer with possible digit(s) before extension
+  $re64 =  @{$true="(WIN|Win)\-(LPv\d+\.\d+\.\d+|Conda|x64)(\-|_)($portable)(\-|.)?(\d+)?(\.$ext)$";$false="(WIN)\-x64\-($portable)(\-|.)?(\d)?\.$ext$"}[( $kind -match 'dev' )]
   $url32 = $download_page.Links | ? href -match $re32 | select -first 1 -expand href
   $url64 = $download_page.Links | ? href -match $re64 | select -first 1 -expand href
+# Logic to find filler value from url
+  $fill = (($url64.Split('\/'))[-2]).Split('.')[-1]
+  if ( (!([string]::IsNullOrEmpty($fill) ) ) -and ($fill -is [int]) ) {
+      Write-Warning "fill -$fill-"
+      if ($fill -ge $filler) { $filler = $fill }
+  }
   $checking = (($url64.Split('\/'))[-1]) -replace($re64,'') -replace('\-','.'); $version = ( Get-Version $checking ).Version
+  # Version Matching for all kind except dev since it is only 64bit
+  if ($kind -ne 'dev'){ 
+      $32bitchk = (($url32.Split('\/'))[-1]) -replace($re32,'') -replace('\-','.'); $ver32 = ( Get-Version $32bitchk ).Version
+	  if ($ver32 -ne $version) {
```

Comment:
This will also set the 32bit URL to null for any type of release, not just dev releases.

If going with this route, we should drop support for 32bit completely and always have `url` set to `''` in the `chocolateyInstall.ps1` file.

I am all for dropping 32bit support.

Reply:
I can understand that, and we can wait with the regard of removing the 32bit for now.
However, can you please update the Readme.md file with a `Note` section that the 32bit and 64bit edition is mismatched?

Then we can get out a new fix version of the package.

Reply:
I will work on the readme update.